Network/RequestBuilder.swift

```diff
@@ -0,0 +1,32 @@
+//
+//  RequestBuilder.swift
+//  WeatherDemoApp
+//
+//  Created by ahmed on 04/04/2024.
+//
+
+import Foundation
+
+extension WeatherFetcher {
+    struct OpenWeatherAPI {
+        static let scheme = "https"
+        static let host = "api.openweathermap.org"
+        static let path = "/data/2.5"
+        static let key = "4c6eb36cdcfd3de4bddb06d5b9b4b760"
+    }
+    
+    // MARK: City weather info
+    func makeWeatherInfoComponents(withCity city: String) -> URLComponents {
```

Comment:
I would like to see when you add two more endpoints, how you will handle the code duplication?